skopt/parameter.py

```diff
@@ -38,178 +37,156 @@ def inverse_transform(self, values):
         return 10**np.asarray(values)
 
 
-class Log(TransformerMixin):
-    """Natural logarithm transform."""
-    def fit(self, values):
-        return self
+class Distribution:
 
-    def transform(self, values):
-        return np.log(values)
+    @abc.abstractmethod
```

Comment:
Do we really need `abc` and stuff? A simple `raise NotImplementedError` wouldnt be enough?

(This is an open question, I have no strong opinion against one or the other.)